### PR TITLE
Delete Garden Plan - first garden

### LINKED ISSUE #20: User deletes a garden plan
## User deletes a garden plan

As a user
In order to remove an unwanted garden plan
I want to delete my garden plans

Acceptance Criteria:
- User must be logged in
- User must have created at least one garden plan

Usage:
- From home screen, user clicks on the garden plan in question
- User selects "Delete This Garden" from the bottom of the "Design" tab
- User is prompted to confirm the deletion
- The user sees a confirmation message
- User is redirected to the home page after deletion
- User remains on the current page if they choose not to delete the plan

### PR DESCRIPTION
**Pull request by marisajackson**
resolves #20 Deletion of first garden successfully implemented including confirmation prompt and flash notice.

Pending tests for multiple gardens.

### CODE REVIEW

File: config/database.yml.example [hunk 1]
```diff
@@ -1,85 +0,0 @@
-# PostgreSQL. Versions 8.2 and up are supported.
```
Comment (elizamarcum) [inline config/database.yml.example:1]: This file shouldn't have been deleted..

File: spec/features/user_deletes_garden_plan_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+# From home screen, user clicks on the garden plan in question
+# User selects "Delete This Garden" from the bottom of the "Design" tab
+# User is prompted to confirm the deletion
+# The user sees a confirmation message
+# User is redirected to the home page after deletion
+# User remains on the current page if they choose not to delete the plan
+
+feature "User deletes a garden plan" do
+	background do
```
Comment (elizamarcum) [inline spec/features/user_deletes_garden_plan_spec.rb]: Please remove the hard tabs in this file and turn on soft tabs in your text editor.

File: spec/features/user_deletes_garden_plan_spec.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+		@user = Fabricate(:user)
+		login_as @user
+	end
+
+	scenario "deleting your first garden" do
+		click_button "Set up your First Garden"
+		click_button "Delete This Garden"
+		page.should have_content("Your garden has been deleted.")
+		current_path.should == root_path
```
Comment (elizamarcum) [inline spec/features/user_deletes_garden_plan_spec.rb]: This test should verify that the garden is no longer present.  As it is, it is only testing that the correct messages are shown.

I would structure this way:

```
Fabricate(:garden, user: @user, name: "The Best Garden")
visit '/'
page.should have_content("The Best Garden")
click_link "The Best Garden"
click_button "Delete This Garden"
click "Confirm Deletion"
page.should have_content("Your garden has been deleted.")
current_path.should == root_path
page.should_not have_content("The Best Garden")
```

You also need a scenario to cover if the user doesn't confirm the deletion.

**Comment by marisajackson** [comment]
@elizabrock Implemented all of the requested fixes.

File: app/controllers/gardens_controller.rb [hunk 4]
```diff
@@ -9,4 +9,11 @@ def create
   def edit
     @garden = Garden.find(params[:id]) # <--- This lets people load gardens that aren't theirs!! When we implement edit, we must fix this vulnerability!
   end
+
+  def destroy
+  	@garden = Garden.find(params[:id])
```
Comment (elizamarcum) [inline app/controllers/gardens_controller.rb:14]: This is using hard tabs.

File: config/routes.rb [hunk 5]
```diff
@@ -8,7 +8,8 @@
   resources :categories, except: :destroy do
     resources :varieties, only: [:new, :create, :show]
   end
-  resources :gardens, only: [:index, :create, :edit]
+
+  resources :gardens, only: [:index, :create, :edit, :destroy]
```
Comment (elizamarcum) [inline config/routes.rb:12]: At this point, the `only` can be changed to `except: [:new, :update]`

**Comment by elizamarcum** [comment]
@marisajackson and @ajmiddleton I'm going to merge this and fix the small issues I noted above.

**Pull request merged**